drivers/scd30/scd30.c

```diff
@@ -0,0 +1,379 @@
+    /* Check if input is valid */
+    if (!_is_valid_interval(interval)) {
+        DEBUG("[scd30] Interval value out of range\n");
+        return SCD30_INVALID_VALUE;
+    }
+    if (!_is_valid_press_comp(apc)) {
+        DEBUG("[scd30] Ambient pressure compensation value out of range\n");
+        return SCD30_INVALID_VALUE;
+    }
+
+    ret = scd30_set_param(dev, SCD30_INTERVAL, *interval);
+    if (ret != 0) {
+        return ret;
+    }
+    ret = scd30_set_param(dev, SCD30_START, *apc);
+    if (ret != 0) {
+        return ret;
+    }
+    return SCD30_OK;
+}
+
+int8_t scd30_stop_measurements(scd30_t *dev)
+{
+    uint8_t buffer[2];
+    uint16_t cmd = SCD30_STOP;
+    int ret;
+
+    buffer[0] = (uint8_t)(cmd >> 8);
+    buffer[1] = (uint8_t)cmd;
+    i2c_acquire(SCD30_I2C);
+    ret = i2c_write_bytes(SCD30_I2C, SCD30_I2C_ADDRESS, buffer, 2, 0);
+    i2c_release(SCD30_I2C);
+    if (ret != 0) {
+        DEBUG("[scd30]: Error stopping measurements.\n");
+    }
+    return SCD30_OK;
+}
+
+/**
+ *      intern function to read data
+ *
+ *      @param  dev      device
+ *		@param  result	 struct in which data read is stored
+ *      @return          SCD30_OK on success
+ */
+static int8_t _scd30_read_data(scd30_t *dev, scd30_measurement_t *result)
+{
+    uint8_t buffer[18];
+    int ret = 0;
+    uint16_t cmd = SCD30_DATA;
+
+    buffer[0] = (uint8_t)(cmd >> 8);
+    buffer[1] = (uint8_t)cmd;
+    i2c_acquire(SCD30_I2C);
+    ret = i2c_write_bytes(SCD30_I2C, SCD30_I2C_ADDRESS, buffer, 2, 0);
+    if (ret != 0) {
+        i2c_release(SCD30_I2C);
+        return ret;
+    }
+    ret = i2c_read_bytes(SCD30_I2C, SCD30_I2C_ADDRESS, buffer, 18, 0);
+    if (ret != 0) {
+        DEBUG("[scd30] read_data: ret %i.\n", ret);
+        i2c_release(SCD30_I2C);
+        return ret;
+    }
+    i2c_release(SCD30_I2C);
+
+    if (_scd30_crc_check(buffer, 2) != 0) {
+        return SCD30_CRC_ERROR;
+    }
+    if (_scd30_crc_check(buffer + 3, 2) != 0) {
+        return SCD30_CRC_ERROR;
+    }
+    if (_scd30_crc_check(buffer + 6, 2) != 0) {
+        return SCD30_CRC_ERROR;
+    }
+    if (_scd30_crc_check(buffer + 9, 2) != 0) {
+        return SCD30_CRC_ERROR;
+    }
+    if (_scd30_crc_check(buffer + 12, 2) != 0) {
+        return SCD30_CRC_ERROR;
+    }
+    if (_scd30_crc_check(buffer + 15, 2) != 0) {
+        return SCD30_CRC_ERROR;
+    }
+
+    result->co2_concentration = _raw_val_to_float(&buffer[0]);
+    result->temperature = _raw_val_to_float(&buffer[6]);
+    result->relative_humidity = _raw_val_to_float(&buffer[12]);
+
+    return SCD30_OK;
+}
+
+/**
+ *      check if crc is valid (Assuming crc value to compare with is stored at
+ *      end of buffer)
+ *
+ *      @param  buff  buffer of which crc value has to be checked
+ *      @param  len   length of buffer
+ *      @return       SCD30_OK on success
+ */
+static int _scd30_crc_check(uint8_t *buff, uint8_t len)
+{
+    if (crc8(buff, len, SCD30_CRC_FUNC, SCD30_CRC_START_VAL) != buff[len]) {
```

Comment:
Why not just
```suggestion
    return crc8(buff, len, SCD30_CRC_FUNC, SCD30_CRC_START_VAL) != buff[len];
```

Reply:
To deliver the error to the user in a more universal form where SCD30_OK is delivered on success and SCD30_CRCERROR otherwise.

Reply:
But the result of `_scd30_crc_check()` is never returned directly.

Reply:
It is in: 
```
uint8_t scd30_read_periodic(scd30_t *dev, scd30_measurement_t *result)
```
and 
```
int8_t scd30_get_param(scd30_t *dev, uint16_t param, uint16_t *val)
```

Reply:
I don't see `_scd30_crc_check()` being called in either of those functions.

Reply:
Oh yeah, it is but now I got your point. Sure, will shorten the functions implementation then